docs for minon kafka tuning (#5061)
OpenNMS/opennms: docs/modules/reference/pages/configuration/tuning-kafka.adoc

== Tips for Kafka

* For Kafka RPC, the number of partitions should always be greater than the number of Minions at a location.
When there are multiple locations, the partition count should be greater than the max number of Minions at a location.

* By default, Kafka RPC uses a buffer size of 900KB.
When set to a larger size, they will be split into 900kb chunks.

> Configure max buffer size by setting `org.opennms.core.ipc.rpc.kafka.max.buffer.size` to the desired size in bytes.

Reviewer: ```suggestion
Configure the maximum buffer size by setting `org.opennms.core.ipc.rpc.kafka.max.buffer.size` to the desired size in bytes.
```

* Default time to live (time at which request will expire) is 20000ms (20 seconds).
To change it, configure the system property `org.opennms.core.ipc.rpc.kafka.ttl` in milliseconds.

> == Disable Single Topic for Kafka RPC

Reviewer: ```suggestion
== Disable single topic for Kafka RPC
```

> By default {page-component-title} uses single topic structure.

Reviewer: ```suggestion
By default, {page-component-title} uses single topic structure.
```

== Tips for Kafka

> * For Kafka RPC, the number of partitions should always be greater than the number of Minions at a location.

Author: ```suggestion
* For Kafka RPC, the number of partitions should be a multiple of the number of Minions at a location.
```

Reviewer 2: Does that mean at least a multiple of 2 of the number of minions at a location? For example, if there's only 2 minions at only one location, that'd be at least 4 partitions.

Author: 2 is a multiple of 2.  If there are 2 minions, there should be 2, 4, 6, 8 ... partitions
If there are 3 minions, there should be 3, 6, 9, ... partitions

Reviewer 2: Thanks for the clarification! @mmahacek

> When there are multiple locations, the partition count should be greater than the max number of Minions at a location.

Reviewer 3: The recommendation for the number of partitions should be a multiple of the number of minions in a particular location.

This will allow for a more even distribution of work.

Author: ```suggestion
When there are multiple locations, the partition count should be based on the maximum number of Minions at a location.
```

Reviewer 3: So generally a location gets it's own topic(s), thus you may want to amend it to something like.

"When there are multiple locations, each location will have its own topics. The partition count for all topics assigned to a particular location should be based on a multiple of the maximum number of Minions assigned to that location."

Thoughts?

Author: ```suggestion
When there are multiple locations, the partition count for RPC topics should be based on a multiple of the maximum number of Minions at the associated location.
```
